Index: openvpn-2.2.0/route.c
===================================================================
--- openvpn-2.2.0.orig/route.c	2011-05-10 16:30:14.928001206 +0200
+++ openvpn-2.2.0/route.c	2011-05-10 16:34:41.432001310 +0200
@@ -432,6 +432,8 @@
 
         CLEAR(netlist);		/* init_route() will not always init this */
 
+	CLEAR(netlist);                /* init_route() will not always init this */
+
 	if (!init_route (&r,
 			 &netlist,
 			 &opt->routes[i],
